blockchain/blockchain.go

```diff
@@ -444,6 +447,17 @@ func (bc *BlockChain) StateAt(root common.Hash) (*state.StateDB, error) {
 	return state.New(root, bc.stateCache)
 }
 
+// StateAtWithGCLock returns a new mutable state based on a particular point in time with read lock of the state nodes.
+func (bc *BlockChain) StateAtWithGCLock(root common.Hash) (*state.StateDB, error) {
```

Comment:
I'm a bit worried about this function since the caller of the function should take care of unlocking.

Reply:
Yes. You should pay attention to the use of this feature.
If you are too worried about this, we have the option of moving this lock / error handling code outward.

Reply:
> Yes. You should pay attention to the use of this feature.

Actually, this is what I addressed before. Taking care of unlocking outside of this function is very error-prone.